File: 최재훈/05M05W/프로그래머스_LV1-3진법 뒤집기/source.py
def solution(n):
    answer = 0

    threeDeg = []
    while(True) :
        if 0 == n:
            break
        
        threeDeg.append(int(n % 3))
        n = int(n / 3)

    print(threeDeg)
    threeDeg.reverse()
    multi = 1
    for item in threeDeg:
        answer += (item * multi)
        multi *= 3

    print(answer)

    return answer

n = 45
solution(n)

n = 125
solution(n)

n = 100000000
solution(n)

File: 최재훈/06M03W/프로그래머스_LV1-같은 숫자는 싫어/source.cpp
#include <vector>
#include <iostream>

#define MAX_SIZE    1000000

using namespace std;

vector<int> solution(vector<int> arr)
{
    vector<int> answer;

    vector<int>::const_iterator iter = arr.begin();
    int prevNum = -1;
    for (; iter != arr.end(); ++iter)
    {
        if (prevNum != *iter)
        {
            answer.push_back(*iter);
            prevNum = *iter;
        }
    }

    return answer;
}

int main()
{
    vector<int> arr = { 1,1,3,3,0,1,1 };
    vector<int> ans = solution(arr);

    for (int item : ans)
    {
        cout << item << " ";
    }
    cout << endl;

    return 0;
}